engine/runtime/src/main/java/org/enso/interpreter/runtime/data/hash/HashMapInsertAllNode.java

```diff
@@ -51,16 +51,16 @@ EnsoHashMap doEnsoHashMaps(
     if (otherSize == 0) {
       return self;
     }
-    var mapBuilder = EnsoHashMapBuilder.createWithCapacity(selfSize + otherSize);
+    var exactCapacity = selfSize + otherSize;
+    var enoughCapacity = exactCapacity + exactCapacity / 2;
+    var mapBuilder = EnsoHashMapBuilder.createWithCapacity(enoughCapacity);
 
-    var selfMapBuilder = self.getMapBuilder(frame, true, hashCodeNode, equalsNode);
-    var selfEntriesIt = selfMapBuilder.getEntriesIterator(selfMapBuilder.generation());
+    var selfEntriesIt = self.getEntriesIterator(frame, hashCodeNode, equalsNode);
```

Comment:
One cannot use _builder_'s `generation()`, @Akirathan - it may be ahead of `EnsoHashMap.generation`. Fixed in 21d9bc3